src/app/CommandHandlerImpl.cpp

```diff
@@ -135,16 +135,7 @@ CHIP_ERROR CommandHandlerImpl::TryAddResponseData(const ConcreteCommandPath & aR
 
     TLV::TLVWriter * writer = GetCommandDataIBTLVWriter();
     VerifyOrReturnError(writer != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    auto context = GetExchangeContext();
-    if (context && context->HasSessionHandle())
-    {
-        ReturnErrorOnFailure(aEncodable.EncodeTo(*writer, TLV::ContextTag(CommandDataIB::Tag::kFields),
-                                                 context->GetSessionHandle()->GetFabricIndex()));
-    }
-    else
-    {
-        ReturnErrorOnFailure(aEncodable.EncodeTo(*writer, TLV::ContextTag(CommandDataIB::Tag::kFields)));
-    }
+    ReturnErrorOnFailure(aEncodable.EncodeTo(*writer, TLV::ContextTag(CommandDataIB::Tag::kFields), GetAccessingFabricIndex()));
```

Comment:
Can discuss https://github.com/project-chip/connectedhomeip/issues/39001 here too

i.e. ideally as we're encoding a response command payload, we'd be able to stuff the fabric index for the current session... but not sure if there's a way?